underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/page/TopLevelElementRenderer.kt

```diff
@@ -224,8 +224,8 @@ class TopLevelElementRenderer(private val linker: Linker, private val sectionRen
 
         sections.add(sectionRenderer.renderParameters(function))
 
-        if (function.documentation?.returnValue?.isNotEmpty() == true) {
-            sections.add(sectionRenderer.renderReturnValue(function, function.documentation.returnValue))
+        if (function.documentation?.returnValue?.isNotEmpty()!!) {
```

Comment:
Could you walk me through why using the !! operator is better in this specific if statement?

Reply:
It is a non-null asserted call. The !! is required to make sure, that the returnValue object exists.
[https://kotlinlang.org/docs/reference/null-safety.html#the--operator](https://kotlinlang.org/docs/reference/null-safety.html#the--operator)

Reply:
Isn't that the job of the safe call operator? If any of the properties is null, the whole chain becomes null, so the if statement becomes false. This is the same behavior as above, becuase null == true is always false. Or am I missing something along the lines?

Reply:
Thanks guys, this was also a nice catch! @battila7 your comment made everything clear.

Reply:
I'm glad we had this issue, since I think, all of us has a little bit better understanding of the `!!` operator from now on.